src/libraries/Common/src/Interop/Browser/Interop.Runtime.NativeAOT.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.JavaScript;
+
+internal static partial class Interop
+{
+    // WARNING: until https://github.com/dotnet/runtime/issues/37955 is fixed
+    // make sure that the native side always sets the out parameters
+    // otherwise out parameters could stay un-initialized, when the method is used in inlined context
+    internal static unsafe partial class Runtime
+    {
+#pragma warning disable SYSLIB1054
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern void ReleaseCSOwnedObject(IntPtr jsHandle);
+        [DllImport("*", EntryPoint = "mono_wasm_bind_js_function")]
+        public static extern unsafe void BindJSFunction(string function_name, int function_name_length, string module_name, int module_name_length, void* signature, out IntPtr bound_function_js_handle, out int is_exception);
```

Comment:
To come back to the marshalling point: this should use `[LibraryImport("*", EntryPoint = "mono_wasm_bind_js_function", StringMarshalling = StringMarshalling.Utf16)]`, with the JS side using `UTF16ToString` (which will need to be included into `EmccExportedRuntimeMethod`, I guess).